### DISCUSSION TITLE
Adding JavaScript components for advanced functionality?

### DISCUSSION THREAD

**Discussion by binaryfire** [filamentphp/filament#637] [Help] [answered]
Hi guys

I'm interested in using Filament as an alternative to a custom Vue.js dashboard. But I need to use CKEditor and some other JS components in certain admin views for advanced functionality.

Is there an easy way of doing this?

Thanks!

**Comment by farhad-rohani** [comment]
yes i have this problem

**Comment by danharrin** [comment, marked as answer]
We have some documentation about building custom form components - is this what you're looking for? https://filamentphp.com/docs/2.x/forms/fields#building-custom-fields

We currently have Trix in the core & TinyMCE editor in a plugin, it should be simple to add others like CKEditor, especially if you've used Alpine before :)

**Reply by forexknight** [reply]
Author is asking about custom js and you are sending "documentation" which is actually not saying anything about custom js and is just ordinary example...

**Reply by danharrin** [reply]
At the moment, the best way to learn about how to use JS in field is to look at all the fields that are pre-built and are included with Filament. They are all good examples.

**Reply by MitichPavel** [reply]
@danharrin 
https://filamentphp.com/docs/2.x/admin/plugins#scripts
Maybe this part of documentation about custom plugin and scripts for it will be useful for you?

**Reply by scottw-finao** [reply]
> Author is asking about custom js and you are sending "documentation" which is actually not saying anything about custom js and is just ordinary example...

Such RTFM answers are all I have received from anything 'filament' related also

**Reply by scottw-finao** [reply]
> At the moment, the best way to learn about how to use JS in field is to look at all the fields that are pre-built and are included with Filament. They are all good examples.

Good answers include sufficient details to know context and/or point to working examples with complete code. Filament documentation is fragmented and incomplete. Half the time it doesn't even tell you what context a given code snippet comes from or goes into.  What is the point of including documentation at all if the pat answer is 'go spelunk working code'?

**Comment by fariqueparammel** [comment]
@danharrin hey, am a newbie developer so  what about using renderhooks , i did it once but now its not working am i missing something is there any command to to run after providing it in provider ? am using @vite directive and and laready run build am i missign something?
